scripts/generate_docs.sc

```diff
@@ -0,0 +1,147 @@
+    case _ => false
+  }
+}
+
+def notIgnoredBranch(branchVersion: String) = !ignoredReleaseBranchesVersions.contains(branchVersion)
+
+def getTheLastTagVersion(tags: Seq[String]) = tags.last.drop(1)
+
+/**
+  * getting 3 latest release versions and corresponding latest tags
+  *
+  * example: List[(String, String)] = List(("1.3", "1.3.14"), ("1.4", "1.4.11"), ("1.5", "1.5.6"))
+  */
+val docsTargetVersions = listAllTagsInOrder
+  .filter(isReleaseTag)
+  .groupBy(releaseBranchVersion)
+  .filterKeys(notIgnoredBranch)
+  .mapValues(getTheLastTagVersion)
+  .toList.sortBy(_._1)
+  .takeRight(3)
+
+val (latestReleaseBranch, latestReleaseVersion) = docsTargetVersions.last
+
+val buildDir = makeTmpDir()
+
+val docsDir = pwd/up/"docs"
+
+val rootDocsDir = buildDir / 'docs
+
+def generateDocsForVersion(docsPath: Path, version: String, outputPath: String = "_site"): Unit = {
+  %('bundle, "exec", s"jekyll build --config _config.yml,_config.$version.yml -d $outputPath/$version/")(docsPath)
+}
+
+def branchForTag(version: String) = {
+  s"tags/v$version"
+}
+
+// step 1: copy docs/docs to the respective dirs
+
+def checkoutDocsToTempFolder() = {
+  docsTargetVersions foreach { case (releaseBranchVersion, tagVersion) =>
+    val tagName = branchForTag(tagVersion)
+    %git('checkout, s"$tagName")
+    val tagBuildDir = buildDir / releaseBranchVersion
+    mkdir! tagBuildDir
+    println(s"Copying $tagVersion docs to: $tagBuildDir")
+    cp.into(docsDir / 'docs, tagBuildDir)
+    cp.into(docsDir / '_layouts / "docs.html", tagBuildDir)
+    println(s"Docs folder for $releaseBranchVersion is copied to ${tagBuildDir / "docs"}")
+  }
+}
+
+// step 2: generate the default version (latest release tag)
+
+def generateTopLevelDocs() = {
+  %git('checkout, branchForTag(latestReleaseVersion))
+
+  println(s"Copying docs for $latestReleaseVersion into $buildDir")
+  cp.into(docsDir, buildDir)
+
+  println("Cleaning previously generated docs")
+  rm! rootDocsDir / '_site
+
+  println(s"Generating root docs for $latestReleaseVersion")
+  %("bundle", "install", "--path", s"vendor/bundle")(rootDocsDir)
+  %('bundle, "exec", s"jekyll build --config _config.yml -d _site")(rootDocsDir)
+}
+// step 3: generate docs for other versions
+def generateVersionedDocs() {
+  println(s"Generating docs for versions ${docsTargetVersions.map(_._2).mkString(", ")}")
+  docsTargetVersions foreach { case (releaseBranchVersion, tagVersion) =>
+    println("Cleaning docs/docs")
+    rm! rootDocsDir / 'docs
+    println(s"Copying docs for $tagVersion to the docs/docs folder")
+    cp.into(buildDir / releaseBranchVersion / 'docs, rootDocsDir)
+    cp.over(buildDir / releaseBranchVersion / "docs.html", rootDocsDir / '_layouts / "docs.html")
+    println(s"Generation docs for $tagVersion")
+    write.over(rootDocsDir / s"_config.$releaseBranchVersion.yml", s"baseurl : /marathon/$releaseBranchVersion")
+    generateDocsForVersion(rootDocsDir, releaseBranchVersion)
+  }
+}
+
+def buildDocs() = {
```

Comment:
I would rather see a `@main` and make this script runnable like our other scripts.